packages/wasm/crate/src/asset.rs

```diff
@@ -3,17 +3,17 @@ use penumbra_asset::asset::Id;
 use penumbra_proto::DomainType;
 use wasm_bindgen::prelude::*;
 
-/// generate the appropriate binary inner id for a binary-serialized protobuf
+/// generate the appropriate AssetId for a binary-serialized protobuf
 /// `AssetId` potentially containing an `altBaseDenom` or `altBech32m` string
 /// field
 ///
 /// Arguments:
 ///     input_id_bin: `Uint8Array` representing a binary-serialized `AssetId`
 ///
 /// Returns:
-///     `Uint8Array` representing the literal inner id
+///     `Uint8Array` representing a binary-serialized `AssetId`
 #[wasm_bindgen]
-pub fn get_asset_id_inner(input_id_bin: &[u8]) -> WasmResult<Vec<u8>> {
+pub fn get_asset_id(input_id_bin: &[u8]) -> WasmResult<Vec<u8>> {
     let input_id = Id::decode(input_id_bin)?;
-    Ok(input_id.to_bytes().to_vec())
```

Comment:
In general, we should be relying on the buf build types and not rely upon the inner implementation details of the type. The former function specifically gave the bytes for the inner field. The fix simply just encodes to bytes for the entirety of the type.